src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java

```diff
@@ -0,0 +1,461 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.isf.mortuarystays.gui;
+
+
+import java.util.EventListener;
+import java.awt.AWTEvent;
+import java.awt.BorderLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.EventListenerList;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+import org.isf.generaldata.MessageBundle;
+import org.isf.menu.manager.Context;
+import org.isf.mortuarystays.manager.MortuaryStayManager;
+import org.isf.mortuarystays.model.MortuaryStay;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.gui.OHServiceExceptionUtil;
+import org.isf.utils.jobjects.MessageDialog;
+import org.isf.utils.jobjects.VoLimitedTextField;
+
+public class MortuaryStaysEdit extends JDialog {
+    private static final long serialVersionUID = 1L;
+    private EventListenerList mortuaryStaysListeners = new EventListenerList();
+
+    public interface MortuaryStaysListener extends EventListener {
+
+        void mortuaryStaysUpdated(AWTEvent e);
+
+        void mortuaryStaysInserted(AWTEvent e);
+    }
+
+    public void addMortuaryStaysListener(MortuaryStaysListener l) {
+        mortuaryStaysListeners.add(MortuaryStaysListener.class, l);
+    }
+
+    public void removeMortuaryStaysListener(MortuaryStaysListener listener) {
+        mortuaryStaysListeners.add(MortuaryStaysListener.class, listener);
+    }
+
+    private void fireMortuaryStaysInserted() {
+        AWTEvent event = new AWTEvent(new Object(), AWTEvent.RESERVED_ID_MAX + 1) {
+
+            private static final long serialVersionUID = 1L;
+        };
+
+        EventListener[] listeners = mortuaryStaysListeners.getListeners(MortuaryStaysEdit.MortuaryStaysListener.class);
+        for (EventListener listener : listeners) {
+            ((MortuaryStaysEdit.MortuaryStaysListener) listener).mortuaryStaysInserted(event);
+        }
+    }
+
+    private void fireMortuaryStaysUpdated() {
+        AWTEvent event = new AWTEvent(new Object(), AWTEvent.RESERVED_ID_MAX + 1) {
+
+            private static final long serialVersionUID = 1L;
+        };
```

Comment:
```suggestion
        }
```

Reply:
This is an assignment instruction, and must end with a semicolon